File: src/test/java/application/auth/logout/LogoutControllerTest.java
package application.auth.logout;

import static application.TestUtils.USER_1_SESSION_TOKEN;
import static application.TestUtils.USER_1_USERNAME;
import static org.assertj.core.api.Assertions.assertThat;
import static org.springframework.http.HttpStatus.OK;

import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import com.matag.admin.session.MatagSessionRepository;

import application.AbstractApplicationTest;

public class LogoutControllerTest extends AbstractApplicationTest {
  @Autowired
  private MatagSessionRepository matagSessionRepository;

  @Test
  public void shouldLogoutAUser() {
    // Given
    userIsLoggedIn(USER_1_SESSION_TOKEN, USER_1_USERNAME);

    // When
    var logoutResponse = restTemplate.getForEntity("/auth/logout", String.class);

    // Then
    assertThat(logoutResponse.getStatusCode()).isEqualTo(OK);
    assertThat(matagSessionRepository.count()).isEqualTo(0);
  }

  @Test
  public void shouldLogoutANonLoggedInUser() {
    // When
    var logoutResponse = restTemplate.getForEntity("/auth/logout", String.class);

    // Then
    assertThat(logoutResponse.getStatusCode()).isEqualTo(OK);
  }
}

File: src/test/kotlin/application/auth/logout/LogoutControllerTest.kt
package application.auth.logout

import application.AbstractApplicationTest
import application.TestUtils
import com.matag.admin.session.MatagSessionRepository
import org.assertj.core.api.Assertions
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.HttpStatusCode

class LogoutControllerTest : AbstractApplicationTest() {
    @Test
    fun shouldLogoutAUser() {
        // Given
        loginUser(TestUtils.USER_1_SESSION_TOKEN, TestUtils.USER_1_USERNAME)

        // When
        val logoutResponse = getForEntity("/auth/logout", String::class.java, TestUtils.USER_1_SESSION_TOKEN)

        // Then
        assertThat(logoutResponse.status).isEqualTo(HttpStatus.OK)
        assertThat(matagSessionRepository.count()).isEqualTo(0)
    }

    @Test
    fun shouldLogoutANonLoggedInUser() {
        // When
        val logoutResponse = getForEntity("/auth/logout", String::class.java)

        // Then
        assertThat(logoutResponse.status).isEqualTo(HttpStatus.OK)
    }
}